### PR TITLE
Add dedicated port type to harness design.json

### PR DESCRIPTION
**Pull request by andrewb1999**
Signed-off-by: Andrew Butt <butta@seas.upenn.edu>

Allows a convergence of prjxray roi harnesses and a general partition region in https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1561

### CODE REVIEW

**Comment by andrewb1999** [comment]
@litghost Any idea why this change would cause the zynq 7 database build to fail?

File: minitests/roi_harness/create_design_json.py [hunk 1]
```diff
@@ -46,6 +46,15 @@ def main():
     design_json['info'] = {}
     with open(args.design_txt) as f:
         for d in csv.DictReader(f, delimiter=' '):
+            if d['name'].startswith('dout['):
+                d['type'] = 'in'
```
Comment (litghost) [inline minitests/roi_harness/create_design_json.py:50]: Ins't this backwards for the file format?  dout[x] is an output, not an input right?

Comment (andrewb1999) [reply minitests/roi_harness/create_design_json.py:50]: Ah yes, you're right.  Fixing now.

**Comment by andrewb1999** [comment]
CI looks good now.  Good to go?

**Comment by litghost** [comment]
@andrewb1999 Once CI runs on the new master, a new prjxray-db can be pushed.  Remind @mithro next Monday or Tuesday and they should be able to update it.

**Comment by mithro** [comment]
New db has been pushed.

**Pull request merged**